Clearer student teacher user type selection

TurnerRiley opened the pull request:
This PR makes the abilities of a Student vs. a Teacher clearer on the sign-up page. Instead of a drop-down that says "Student" and "Teacher," there are 2 large buttons that describe each one. This way, teachers are more likely to see that they have all the abilities a student has (including seeing from a student's view) and will actually select the "Teacher" user type. These changes are toggled by the experiment clearerSignUpUserType.

## Screenshots
Here is the original look (experiment off):
![Sign Up Original](https://user-images.githubusercontent.com/56283563/125360413-ab759300-e320-11eb-8ed5-b8ad9cfb2a1a.JPG)

Here is the updated look (experiment on) which can be found by turning on the experiment with [this link](https://staging-studio.code.org/users/sign_up?enableExperiments=clearerSignUpUserType) (and this experiment can be turned off with [this link](https://staging-studio.code.org/users/sign_up?disableExperiments=clearerSignUpUserType))once the branch is merged into staging:
![Aligned buttons](https://user-images.githubusercontent.com/56283563/125360331-9436a580-e320-11eb-8bd3-21a91451d97e.JPG)

Here is the mobile view:
![Mobile view user type](https://user-images.githubusercontent.com/56283563/125360178-520d6400-e320-11eb-8241-df46887fa63a.JPG)

Here is what it looks like when a user tries to sign up without selecting either option (the expected behavior is what is shown where the red text tells the user they need to deliberately pick a user type):
![Error if user does not select user type in variant](https://user-images.githubusercontent.com/56283563/125360317-900a8800-e320-11eb-9031-c78632bc71b2.JPG)

## Links
Spec can be found [here](https://docs.google.com/document/d/10_QiB-v_VFgGfCB4agK1vZ0WrsQR3i3j6qGLHWMwJ_s/edit).

## Testing story
Manually tested by both selecting a Student account in the original and variant versions, and selecting a Teacher account in the original and variant versions. Since there are only 2 user types, there are very few changes other than visual.

## Follow-up work
If this is all good to go in staging, I will create an Optimizely A/B experiment to test its efficacy.

KatieShipley commented on apps/src/sites/studio/pages/devise/registrations/_finish_sign_up.js:
```diff
@@ -37,6 +37,24 @@ $(document).ready(() => {
+    // TO-DELETE ONCE OPTIMIZELY-EXPERIMENT IS COMPLETE (start)
+    let inVariant = false;
+    const queryString = window.location.search.substring(1).split('&');
+    for (let i = 0; i < queryString.length; i++) {
```
You could also use `.find` here rather than looping through the query string array.
ex. `const inVariant = queryString.find(param => param === 'inSignUpUserTypeVariant=true');`

TurnerRiley replied:
Oh good point! I may actually use .includes() instead since that will give me a boolean

KatieShipley replied:
Even better!

hacodeorg commented on apps/src/sites/studio/pages/devise/registrations/_finish_sign_up.js:
```diff
@@ -37,6 +37,19 @@ $(document).ready(() => {
   function init() {
+    // TO-DELETE ONCE OPTIMIZELY-EXPERIMENT IS COMPLETE (start)
+    const queryString = window.location.search.substring(1).split('&');
+    const inVariant = queryString.includes('inSignUpUserTypeVariant=true');
```
You can take a look at these PRs to see how they set and retrieve experiment flags.
https://github.com/code-dot-org/code-dot-org/pull/29335
https://github.com/code-dot-org/code-dot-org/pull/28653

TurnerRiley replied:
Thank you! I switched it to follow that format.

KatieShipley replied:
Oh very cool! Thanks for the context, @hacodeorg !

daynew commented:
Should probably get the right edge of the inputs to be aligned. Also, could you put a preview screenshot of what this would like like on a smartphone screen?
![Screen Shot 2021-07-12 at 2 16 28 PM](https://user-images.githubusercontent.com/1372238/125357115-b5b28400-e356-11eb-97be-88ea5088b671.png)

This might be a little tough to do, because those empty spaces to the right of the inputs are reserved or error messages. Is there an error message which pops up if student or teacher is not selected? I remember when I was working on the account creation page that we did not default to student or teacher. We wanted the user to make a conscious choice for their account type. Make sure to double check this requirement with @dju90

TurnerRiley commented:
@daynew Thank you for the feedback! I've aligned the buttons and updated the screenshots to show this. I also added one of the mobile view which seems to stay the same and scale properly when I zoom in on the phone. Lastly, I attached a picture of the behavior when the user does not select either button to show that they have to deliberately click one.

dju90 commented:
Padding on the left seems a little narrow. Can we increase by a few px? (looking at the padding left of the checkbox in "i am a parent/guardian..." for comparison)
